internal/maintenance/reindexer_test.go

```diff
@@ -44,12 +44,24 @@ func TestReindexer(t *testing.T) {
 		svc := NewReindexer(archetype, &ReindexerConfig{
 			ScheduleFunc: fromNow(500 * time.Millisecond),
 		}, bundle.exec)
+		svc.StaggerStartupDisable(true)
 		svc.TestSignals.Init()
 		t.Cleanup(svc.Stop)
 
 		return svc, bundle
 	}
 
+	runImmediatelyThanOnceAnHour := func() func(time.Time) time.Time {
+		alreadyRan := false
+		return func(t time.Time) time.Time {
+			if alreadyRan {
+				return t.Add(time.Hour)
+			}
+			alreadyRan = true
+			return t.Add(time.Millisecond)
+		}
```

Comment:
does `TickerWithInitialTick` not work here?

Reply:
I thinkkk they're kind of orthogonal. The main purpose of this function is that it explicitly returns something that's got the right signature to be a `ScheduleFunc`.